api/src/main/java/org/openmrs/module/m2sysbiometrics/scheduler/NationalSynchronizationTask.java

```diff
@@ -15,8 +22,39 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(NationalSynchronizationTask.class);
 
+    private NationalSynchronizationFailureService nationalSynchronizationFailureService;
+
+    private RegistrationService registrationService;
+
+    private SearchService searchService;
+
     @Override
     public void execute() {
         LOGGER.info("Executing " + TASK_NAME  + "...");
+        initializeBeans();
+        retryRegistrationFailures();
+    }
+
+    private void initializeBeans() {
+        nationalSynchronizationFailureService = Context.getRegisteredComponent(
+                "nationalSynchronizationFailureService", NationalSynchronizationFailureService.class);
+
+        registrationService = Context.getRegisteredComponent(
+                "registrationService", RegistrationService.class);
+
+        searchService = Context.getRegisteredComponent(
+                "searchService", SearchService.class);
+    }
+
+    private void retryRegistrationFailures() {
+        for (NationalSynchronizationFailure failure : nationalSynchronizationFailureService.findAllRegistrationFailures()) {
+            nationalSynchronizationFailureService.delete(failure);
```

Comment:
I'm not sure if this is a safe approach. What if a registration failure is deleted and then handling fails?

Reply:
shouldn't you check if national server is configured?

Reply:
Should't you extract these operations inner 'for' loop to a new method and make it Transactional

Reply:
@pawelcieszko there's no need to check if the national server is configured. It's checked where needed and if it's not set then synchronization simply fails and stays in the queue.

Reply:
@dserkowski Transactional wouldn't work on private method, however, I'll try to make this whole task transactional.